modules/ztp-image-based-upgrade-prep.adoc

```diff
@@ -0,0 +1,143 @@
+// Module included in the following assemblies:
+// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
+// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ztp-image-based-upgrade-prep_{context}"]
+= Preparing the {sno} clusters for the image-based upgrade
+
+When you deploy the {lcao} on a cluster, the Operator creates an `imageBasedUpgrade` CR.
+You use this CR throughout the upgrade process to move through the different stages.
+
+In the `Prep` stage, the {lcao} ensures that the nodes are ready for the upgrade by checking if the cluster meets certain conditions.
+You can complete this stage well before starting the upgrade.
+
+After you start the `Prep` stage by applying the `imageBasedUpgrade` CR with {cgu-operator}, {cgu-operator} applies a policy that starts the following procedures:
+
+* Installing and configuring the OADP Operator
+* Adding a `DataProtectionApplication` CR that points to the S3 backend
+* Adding a secret for the S3 credentials that are referenced in the `DataProtectionApplication` CR
+
+These resources are deleted after the upgrade is complete or if a rollback is initiated.
+
+.Prerequisites
+
+* You have installed {cgu-operator}.
+* You have installed {lcao}.
+* You have generated a seed image from a compatible seed cluster.
+
+// are there other prereqs?
+
+.Procedure
+
+. Apply the shared container folder `MachineConfig` CR to make `/var/lib/containers` shared between `ostree` updates.
++
+[source,terminal]
+----
+$ oc apply -f <shared_MC_folder>
+----
+
+. Update the `seedImageRef.version` and `seedImageRef.image` fields in the `ImageBasedUpgrade` CR:
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: example-upgrade
+spec:
+  stage: Idle
+  seedImageRef:
+    version: 4.14.0
+    image: <seed_container_image>
+----
+// For telco, we need a sample with extraManifests and oadpContent
+
+. Move the CR to the `Prep` stage:
++
+[source,terminal]
+----
+$ oc patch imagebasedupgrades.lca.openshift.io upgrade -p='{"spec": {"stage": "Prep"}}' --type=merge -n openshift-lifecycle-agent
+----
+
++
+The {lcao} checks for the health of the cluster and pulls the seed image to the target cluster.
+Then, the Operator pre-caches all the required images.
+
+// What else, if anything?
+
+.Verification
+
+. Check that the new stateroot with the target version exists:
++
+[source,terminal]
+----
+$ rpm-ostree status -v
```

Comment:
I don't think these steps shouldbe part of an end customer workflow. Basically when the status is Prep completed these are done,

Reply:
Good point, these are redundant then. However, I'll leave the IBU CR status check, that could be useful for general monitoring.